From b9678dab44cfb0f1ab4904ee12ac5b3719599b83 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Mon, 23 May 2011 18:34:46 +0200
Subject: [PATCH] Register /bin/true as URI scheme handler for several schemes

Starting with glib 2.28, we don't use gconf to find out which handler
should be used for a URI scheme, and we need to provide a custom MIME
configuration for the gdm user to ensure no default URI scheme handler
is used.
---
 data/Makefile.am                |   10 ++++++++++
 data/mime-dummy-handler.desktop |    6 ++++++
 data/mimeapps.list              |   19 +++++++++++++++++++
 3 files changed, 35 insertions(+), 0 deletions(-)
 create mode 100644 data/mime-dummy-handler.desktop
 create mode 100644 data/mimeapps.list

diff -Nur -x '*.orig' -x '*~' gdm-3.0.0//data/Makefile.am gdm-3.0.0.new//data/Makefile.am
--- gdm-3.0.0//data/Makefile.am	2011-06-06 00:14:07.000000000 +0200
+++ gdm-3.0.0.new//data/Makefile.am	2011-06-06 00:39:16.184147166 +0200
@@ -98,6 +98,8 @@
 	session-setup.entries	\
 	make-dconf-override-db.sh \
 	dconf-profile \
+	mime-dummy-handler.desktop	\
+	mimeapps.list		\
 	$(NULL)
 
 CLEANFILES = 				\
@@ -274,6 +276,14 @@
 	chmod 1750 $(DESTDIR)$(workingdir)/.gconf.mandatory
 	chmod 1640 $(DESTDIR)$(workingdir)/.gconf.mandatory/*.xml
 
+	if test '!' -d $(DESTDIR)$(workingdir)/.local/share/applications; then \
+		$(mkinstalldirs) $(DESTDIR)$(workingdir)/.local/share/applications; \
+		chmod 0755 $(DESTDIR)$(workingdir)/.local/share/applications; \
+		chown gdm:gdm $(DESTDIR)$(workingdir)/.local/share/applications || : ; \
+	fi
+
 	$(INSTALL_DATA) $(srcdir)/dconf-override-db $(DESTDIR)$(sysconfdir)/dconf/db/gdm
 	$(INSTALL_DATA) $(srcdir)/dconf-profile $(DESTDIR)$(sysconfdir)/dconf/profile/gdm
+	$(INSTALL_DATA) $(srcdir)/mime-dummy-handler.desktop $(DESTDIR)$(workingdir)/.local/share/applications/mime-dummy-handler.desktop
+	$(INSTALL_DATA) $(srcdir)/mimeapps.list $(DESTDIR)$(workingdir)/.local/share/applications/mimeapps.list
 
diff -Nur -x '*.orig' -x '*~' gdm-3.0.0//data/mimeapps.list gdm-3.0.0.new//data/mimeapps.list
--- gdm-3.0.0//data/mimeapps.list	1970-01-01 01:00:00.000000000 +0100
+++ gdm-3.0.0.new//data/mimeapps.list	2011-06-06 00:43:00.074475538 +0200
@@ -0,0 +1,19 @@
+[Default Applications]
+x-scheme-handler/file=mime-dummy-handler.desktop
+x-scheme-handler/ftp=mime-dummy-handler.desktop
+x-scheme-handler/ghelp=mime-dummy-handler.desktop
+x-scheme-handler/help=mime-dummy-handler.desktop
+x-scheme-handler/http=mime-dummy-handler.desktop
+x-scheme-handler/https=mime-dummy-handler.desktop
+x-scheme-handler/info=mime-dummy-handler.desktop
+x-scheme-handler/irc=mime-dummy-handler.desktop
+x-scheme-handler/itms=mime-dummy-handler.desktop
+x-scheme-handler/mailto=mime-dummy-handler.desktop
+x-scheme-handler/man=mime-dummy-handler.desktop
+x-scheme-handler/mms=mime-dummy-handler.desktop
+x-scheme-handler/rtp=mime-dummy-handler.desktop
+x-scheme-handler/rtsp=mime-dummy-handler.desktop
+x-scheme-handler/sip=mime-dummy-handler.desktop
+x-scheme-handler/trash=mime-dummy-handler.desktop
+x-scheme-handler/webcal=mime-dummy-handler.desktop
+x-scheme-handler/xmpp=mime-dummy-handler.desktop
diff -Nur -x '*.orig' -x '*~' gdm-3.0.0//data/mime-dummy-handler.desktop gdm-3.0.0.new//data/mime-dummy-handler.desktop
--- gdm-3.0.0//data/mime-dummy-handler.desktop	1970-01-01 01:00:00.000000000 +0100
+++ gdm-3.0.0.new//data/mime-dummy-handler.desktop	2011-06-06 00:40:56.824294772 +0200
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Type=Application
+Name=Dummy URI Handler
+Exec=/bin/true %U
+Terminal=false
+StartupNotify=false
